grafana/grafana
# Tracing: Fix links to traces in Explore

**What this PR does / why we need it**:
The previous implementation of span references only handled linking to a span in the same trace. This update fixes that. In the span links menu and the references table (in the span details), links to the same trace shouldn't navigate, but links to a different trace should. 

**Which issue(s) this PR fixes**:



Fixes #48018

**Special notes for your reviewer**:
How to test:
The demos we use for testing don't create span references, so I just hard-code adding in a span reference in `createSpanLink` which can be used for testing. 
1. In Tempo or Jaeger, search for traces
2. Update the hard-coded references in `createSpanLink` to use one of the traces from 1.
3. View a different trace id and check it links correctly to the same trace and a different trace (check links menu and in span details)
![Screen Shot 2022-06-02 at 9 13 17 AM](https://user-images.githubusercontent.com/12838032/171662166-a1d83457-663b-4ac4-be68-d24a23e8aab8.png)

Todo:
- [x] Test in dashboard
```
    // Add at createSpanLink L 191 (right before get trace links)
    if (span.references.length <= 1) {
      span.references.push({
        refType: 'FOLLOWS_FROM',
        spanID: span.spanID,
        traceID: '55cb11440892807e', // Replace with some other trace that exists locally for testing
      });
      span.references.push({
        refType: 'FOLLOWS_FROM',
        spanID: span.spanID,
        traceID: span.traceID,
        span: {
          operationName: 'Same',
          process: {
            serviceName: 'Service',
          },
        } as any,
      });
    }
```
